File: week6/pset6/sentimental-readability/readability.py
# computes the approximate grade level needed to compre hend some text

from cs50 import get_string


def main():
    # Ask the user for the text
    text = get_text()

    # Count the number of letters in the text
    letters = count_letters(text)

    # Count the number of words in the text
    words = count_words(text)

    # Count the number of sentences in the text
    sentences = count_sentences(text)

    # Calculate the average number of letters per 100 words
    L = letters / words * 100

    # Calculate the average number of sentences per 100 words
    S = sentences / words * 100

    # Calculate the grade level according to the Coleman-Liau formula
    level = calculate_level(L, S)

    # Print the grade level

    # Deal with special cases
    if level < 1:
        print("Before Grade 1")

    elif level >= 16:
        print("Grade 16+")

    else:
        print(f"Grade {round(level)}")


def get_text():
    text = get_string("Text: ")

    return text


def count_letters(text):
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1

    return letters


def count_words(text):
    words = 0

    if len(text) > 0:
        words += 1

    for char in text:
        if char == ' ':
            words += 1

    return words


def count_sentences(text):
    sentences = 0

    for char in text:
        if char == '.' or char == '!' or char == '?':
            sentences += 1

    return sentences


def calculate_level(L, S):
    level = 0.0588 * L - 0.296 * S - 15.8

    return level


main()

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
   
    int count_letters(string text);
    int count_sentences(string text);
    int count_words(string text);

    // Get the text from the user
    string text = get_string("Text: ");
    
    // Calculate the number of words
    int w = count_words(text);
    
    // Calculate the average of letters per 100 words
    float l = (float) count_letters(text) / (float) w * 100;

    // Calculate the average of sentences per 100 words
    float s = (float) count_sentences(text) / (float) w * 100;

    // Recall and the Coleman-Liau index 
    float indexf = 0.0588 * l - 0.296 * s - 15.8;
    // Round the index
    int index;
    if (indexf - (int) indexf >= 0.5)
    {
        index = (int) indexf + 1;
    }
    else
    {
        index = indexf;
    }

    //Print the index       
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}


int count_letters(string text)
{
    int l = 0;
    for (int i = 0; i < strlen(text); i++)
    {

        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            l++;
        }
    }
    return l;
}

int count_words(string text)                    
{
    int w = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((isspace(text[i]) != 0))
        {
            w++;
        }
    }
    return w;
}

int count_sentences(string text)
{
    int s = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == '.') || (text[i] == '?') || (text[i] == '!'))
        {
            s++;
        }
    }
    return s;
}
